libraries/aws/jobs/src/jobs.c

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "jobs.h"
+
+typedef enum
+{
+    true = 1, false = 0
+} bool_;
+
+/**
+ * @brief Table of topic API strings in JobsTopic_t order
+ */
+static const char * const apiTopic[] =
+{
+    JOBS_API_JOBSCHANGED,
+    JOBS_API_NEXTJOBCHANGED,
+    JOBS_API_GETPENDING JOBS_API_SUCCESS,
+    JOBS_API_GETPENDING JOBS_API_FAILURE,
+    JOBS_API_STARTNEXT JOBS_API_SUCCESS,
+    JOBS_API_STARTNEXT JOBS_API_FAILURE,
+    JOBS_API_DESCRIBE JOBS_API_SUCCESS,
+    JOBS_API_DESCRIBE JOBS_API_FAILURE,
+    JOBS_API_UPDATE JOBS_API_SUCCESS,
+    JOBS_API_UPDATE JOBS_API_FAILURE,
+};
```

Comment:
Instead of indexing into an array with the JobsTopic_t enum, why not write a function to return a string for a JobsTopic_t input? That way these values are not dependent on the order of the enums, and changing the order in one place wouldn't require changes in the others.

Reply:
The order is useful.  Both Jobs_GetTopic() and matchApi() take advantage of it.  I added a comment for emphasis.